src/components/Map.js

```diff
@@ -29,9 +35,9 @@ class Map extends Component {
     });
   }
 
-  shouldComponentUpdate(nextProps) {
+  componentDidUpdate(nextProps) {
+    let map = this.state.map;
```

Comment:
This variable needs to be volatile, because the map object changes whenever we manipulate it graphically.